lte/gateway/python/magma/pipelined/ifaces.py

```diff
@@ -31,8 +32,33 @@ def monitor_ifaces(iface_names):
         yield from asyncio.sleep(POLL_INTERVAL_SECONDS)
 
 
-def get_mac_address(interface_name: str) -> str:
-    if_addresses = netifaces.ifaddresses(interface_name)[netifaces.AF_LINK]
-    if not if_addresses or not if_addresses[0].get('addr'):
-        raise ValueError(f"No mac address found for interface {interface_name}")
-    return if_addresses[0]['addr']
+def get_mac_address(
+        interface: Optional[str] = None,
+        ip4: Optional[str] = None,
+        ip6: Optional[str] = None,
+) -> str:
+    if interface:
+        ifaddress = netifaces.ifaddresses(interface)
+        if netifaces.AF_LINK in ifaddress:
+            return ifaddress[netifaces.AF_LINK][0]['addr']
+        raise ValueError(f"No mac address found for interface {interface}")
+    elif ip4:
+        return _get_mac_from_ip(ip4, "ip4")
+    elif ip6:
+        return _get_mac_from_ip(ip6, "ip6")
+    raise ValueError("Must specify either interface or ip4 or ip6")
+
+
+def _get_mac_from_ip(ip: str, ipv: str) -> str:
```

Comment:
It seems the function we're trying to replace was also able to find addresses for remote IPs. Do we need that feature?

Reply:
Isn't that what `get_gw_mac_address` is implementing?